src/iframe-helper.js

```diff
@@ -32,9 +32,17 @@ import {parseUrl} from './url';
 export function listen(iframe, typeOfMessage, callback, opt_is3P) {
   assert(iframe.src, 'only iframes with src supported');
   const origin = parseUrl(iframe.src).origin;
-  const win = iframe.ownerDocument.defaultView;
+  let win = iframe.ownerDocument.defaultView;
   const sentinel = getSentinel_(opt_is3P);
-  const listener = function(event) {
+  let unlisten;
+  let listener = function(event) {
+    // If this iframe no longer has a contentWindow is was removed
+    // from the DOM. Unlisten immediately as we can never again receive
+    // messages for it (
+    if (!iframe.contentWindow) {
```

Comment:
There are no message buses per iframe, so while this might never happen, any message from any other child iframe or parent container would invoke this listener.

This isn't reliable, but it is definitely better than letting these leak. It is more defense in depth, than something we should rely on.